homework/list.js

```diff
@@ -1,79 +1,83 @@
 'use strict';
 
-(function() {
+(function(exports) {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+  var ListManager = function() {
+    var _listNoteContent = [];
+    var _wrapper = null;  };
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
-  }
+  ListManager.prototype = {
+    start() {
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this.fetchList().then((function(data) {
```

Comment:
I recommend to use Arrow function: `() => { }` instead of binding this manually.
See:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#Lexical_this